crates/subspace-node/src/bin/subspace-node.rs

```diff
@@ -74,7 +74,39 @@ fn main() -> std::result::Result<(), Error> {
     let cli = Cli::from_args();
 
     if !cli.secondarychain_args.is_empty() {
-        println!("Unimplemented: Run an executor with an embedded primary full node");
+        let secondary_chain_cli =
```

Comment:
I don't understand why is this all before `match` statement, shouldn't it be in `None => {` branch os `match` statement instead?

Also `Subcommand::PurgeChain` command wasn't updated the way it behaved in `subspace-executor` accordingly and probably some others that I don't recall immediately.

I was expecting that primary chain would start just like before and in its `.map(|full| {` we'd check for `cli.secondarychain_args` and run it before calling `full.network_starter.start_network();`. Right now it doesn't seem like  networking is started at all.

Reply:
>I don't understand why is this all before match statement, shouldn't it be in None => { branch os match statement instead?

That's reasonable, updated.

>Also Subcommand::PurgeChain command wasn't updated the way it behaved in subspace-executor accordingly and probably some others that I don't recall immediately.

All executor-related sub-commands are TODO for now.

>I was expecting that primary chain would start just like before and in its .map(|full| { we'd check for cli.secondarychain_args and run it before calling full.network_starter.start_network();. Right now it doesn't seem like networking is started at all.

When running an executor, the runner has to be created using `SecondaryChainCli` in which the primary node will be started instead of creating a runner using the primary node arguments.

Primary networking is started here:
https://github.com/subspace/subspace/blob/fba21ad1355e475a0ddf63fb747464cc04a524be/cumulus/node/src/service.rs#L308

Reply:
I see. In future we'll probably want to have it the other way around and start networking outside of that.